service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOController.java

```diff
@@ -77,7 +77,8 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestHeader("autho
         final Map<String, Object> data = caseDetails.getData();
         final CaseData caseData = mapper.convertValue(data, CaseData.class);
 
-        Map<String, Object> cmoTemplateData = docmosisTemplateDataGenerationService.getTemplateData(caseData, true);
+        Map<String, Object> cmoTemplateData = docmosisTemplateDataGenerationService.getCaseManagementOrderData(caseData)
+            .toMap(mapper);
 
         Document document = getDocument(authorization, userId, cmoTemplateData);
```

Comment:
Not really sure where the best place to put this comment is but thoughts on having `AbstractDocmosisOrder` extend a base class called something like `AbstractDocmosisData` or something like that which has the default implementation of the toMap method
```java
public abstract class AbstractDocmosisData {
    public Map<String, Object> toMap(ObjectMapper mapper) {
        return mapper.convertValue(this, new TypeReference<>() {});
    }
}
```
and the the subsequent classes can call this method and perform additional logic
```java
public class AbstractDocmosisOrder extends AbstractDocmosisData {
    // fields...

    public Map<String, Object> toMap(ObjectMapper mapper) {
        Map<String, Object> map = super.toMap(mapper);
        
       // additional logic...

        return map;
    }
}
```
With this we can then have an overloaded the `DocmosisDocumentGeneratorService.generateDocmosisDocument`
to have a method that looks like
``` java
public class DocmosisDocumentGeneratorService {
    // fields and other methods...

    public DocmosisDocument generateDocmosisDocument(AbstractDocmosisData templateData, DocmosisTemplates template) {
        return generateDocmosisDocument(templateData.toMap(mapper), template);
    }
}
```
Whether we would also pass the mapper as a parameter in the method or autowire it into the class is another question as well.

I feel like this would at least from a readability point of view be an improvement as we would no longer be passing around a map that doesn't have that much meaning but instead an object which is more clear as to what it will be doing, it's use, and it's contents.